backend/src/controllers/goal.controller.ts

```diff
@@ -57,21 +57,23 @@ export const getGoal = async (
     if (stringGoal?.includes(goalId)) {
       // Find goal from database
       const goal = await goalService.getGoal(goalId);
-
-      let goalDTO = JSON.parse(JSON.stringify(goal));
+      if (!goal) {
+        res.sendStatus(httpStatus.NOT_FOUND).end();
+        return;
+      }
       const time_now = new Date(Date.now());
-      const new_date_end = new Date();
+      time_now.setUTCHours(0, 0, 0, 0);
 
       // If the date is surpassed and it is a recurring goal, update the current start/end dates accordingly
-      if (goalDTO.date_start > time_now && goalDTO.recurring) {
-        goalDTO.date_start = time_now;
-        goalDTO.date_end = new_date_end.setDate(time_now + goalDTO.duration);
+      if (goal.date_end < time_now && goal.recurring) {
+        goal.date_start = time_now;
+        goal.date_end = new Date(time_now.getTime() + parseInt(goal.duration));
```

Comment:
getTime() is in milliseconds so this line would just set the end date to be a few milliseconds after the start

Could use this logic
```js
goal.date_end = new Date()
goal.date_end.setDate(goal.date_start.getDate() + parseInt(goal.duration))
goal.date_end.setUTCHours(0, 0, 0, 0)
```